app/components/Home.tsx

```diff
@@ -0,0 +1,270 @@
+const SplitMode = 1;
+const PreviewMode = 2;
+
+const modes = [
+  { mode: EditMode, label: "Edit", css: "edit" },
+  { mode: SplitMode, label: "Split", css: "split" },
+  { mode: PreviewMode, label: "Preview", css: "preview" },
+];
+
+const markdownitOptions = {
+  // automatically render raw links as anchors.
+  linkify: true
+};
+
+const MarkdownParser = MarkdownIt(markdownitOptions)
+  .use(require('markdown-it-footnote'))
+  .use(require('markdown-it-task-lists'))
+  .use(require('markdown-it-highlightjs'));
+
+type AppProps = {
+  text?: string,
+  mode: any;
+  platform?: any,
+}
+
+const initialState = {
+  mode: modes[0],
+  text: "",
+}
+const debugMode = false;
+
+if (debugMode) {
+  console.log("home loaded 2");
+}
+let keyMap = new Map();
+
+export class Home extends React.Component<{}, AppProps> {
+  editorKit: any;
+  note: any;
+
+  constructor(props: AppProps) {
+    super(props);
+    if (debugMode) {
+      console.log("started");
+    }
+    this.state = initialState;
+  }
+
+  componentDidMount = () => {
+    this.configureEditorKit();
+    this.configureMarkdown();
+    this.configureResizer();
+  }
+
+  configureEditorKit() {
+    if (debugMode){
+      console.log("editor kit loaded");
+    }
+    let delegate = new EditorKitDelegate({
+      setEditorRawText: (text: string) => {
+        this.setState(
+          {
+            text,
+          }, () => {
+            this.updatePreviewText();
+            this.loadSavedMode();
+          });
+      },
+      clearUndoHistory: () => { },
+      //getElementsBySelector: () => [],
+    });
+
+    this.editorKit = new EditorKit({
+      delegate: delegate,
+      mode: 'plaintext',
+      supportsFilesafe: false,
+    });
+  };
+
+  saveNote = () => {
+    this.editorKit.onEditorValueChanged(this.state.text);
+  }
+
+  handleInputChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    const target = event.target;
+    const value = target.value;
+
+    this.setState({
+      text: value
+    }, () => {
+      this.updatePreviewText();
+      this.saveNote();
+    });
+  };
+
+  loadSavedMode = () => {
+    try {
+      let savedMode = this.editorKit.internal.componentManager.componentDataValueForKey("mode");
+      if (savedMode) {
+        this.setModeFromModeValue(savedMode);
+      }
+      this.setState({ platform: this.editorKit.internal.componentManager.platform });
+    }
+    finally {
+      return
+    }
+  }
+
+  setModeFromModeValue = (value: number) => {
+    for (const mode of modes) {
+      if (mode.mode == value) {
+        if (debugMode) {
+          console.log("setModeFromModeValue mode: " + mode.mode)
+        }
+        this.setState({ mode: mode });
+        return;
+      }
+    }
+  }
+
+  changeMode = (mode: AppProps) => {
+    this.setState({ mode: mode });
+      if (debugMode) {
+        console.log("changeMode mode: " + mode.mode)
+      }
+      this.editorKit.internal.componentManager.setComponentDataValueForKey("mode", mode.mode);
```

Comment:
Ideally "mode" would be refactored into a named constant at the top of the file so we can take advantage of type checking.

Reply:
I split mode into three different states: https://github.com/sn-extensions/markdown-basic/pull/29/commits/d0de1380b900e7ebee726cf0f8e133e70e7a2d9b Is that OK or do you prefer the named constants/enum?

Reply:
I don't know about this. You decoupled css/labels from their modes. Now we have this.state.css, but there's no clear indication that this is tied to a mode, when it is. I think it's more clear when you just had a mode object that had a css/label property.

Reply:
Enums are implemented as we discussed 👍